test/test-temp-directory.js

```diff
@@ -23,6 +23,10 @@ var badContext = {
   }
 };
 
+// When run through `npm test`, npm will set npm_config_tmp.
+// Unset it to test the no env variable case.
+delete process.env['npm_config_tmp'];
```

Comment:
would it make sense to do this as a test? Should we cache the value in setup and reset in teardown. I'm not a big fan of having the tests leave the system different than when it found it